src/System.Private.ServiceModel/tools/SelfHostedWcfService/Program.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System;
+using System.ServiceModel;
+using System.ServiceModel.Description;
+using System.ServiceModel.Web;
+using WcfService;
+
+namespace SelfHostedWCFService
+{
+    public class SelfHostedWCFService
+    {
+        static int httpPort = 8081;
+        static int httpsPort = 44285;
+        static int tcpPort = 809;
+        static int websocketPort = 8083;
+        static int websocketsPort = 8084;
+
+        static private void GetPorts()
+        {
+            httpPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("httpPort")) ? httpPort : int.Parse(Environment.GetEnvironmentVariable("httpPort"));
+            httpsPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("httpsPort")) ? httpsPort : int.Parse(Environment.GetEnvironmentVariable("httpsPort"));
+            tcpPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("tcpPort")) ? tcpPort : int.Parse(Environment.GetEnvironmentVariable("tcpPort"));
+            websocketPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("websocketPort")) ? websocketPort : int.Parse(Environment.GetEnvironmentVariable("websocketPort"));
+            websocketsPort = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("websocketsPort")) ? websocketsPort : int.Parse(Environment.GetEnvironmentVariable("websocketsPort"));
+        }
+
+        static void Main()
+        {
+            GetPorts();
+
+            string httpBaseAddress = string.Format(@"http://localhost:{0}", httpPort);
+            string httpsBaseAddress = string.Format(@"https://localhost:{0}", httpsPort);
+            string tcpBaseAddress = string.Format(@"net.tcp://localhost:{0}", tcpPort);
+            string websocketBaseAddress = string.Format(@"http://localhost:{0}", websocketPort);
+            string websocketsBaseAddress = string.Format(@"https://localhost:{0}", websocketsPort);
```

Comment:
Not blocking, but let's record this for later discussion.  This "localhost" dependency breaks the dev cross-machine scenario using self-host.  Jason and I use that every day.  I believe the "known requirements" should include not disabling scenarios we actively use today. (For the record, finding and fixing all the dependencies on localhost was quite costly last time we did this for the Bridge.  It's better to avoid hard-coding it in new code.